### PR TITLE
Don't expand parent element if it's the root in single folder workspace when editing an element in explorer.

### LINKED ISSUE #72626: Dangling input elements after renaming top level file element in explorer view in single folder workspace
Steps to Reproduce:

1. Invoke rename on a top level file element in explorer view in single folder workspace
2. Check in developer tools that the div `monaco-tl-contents` for that file row contains multiple invisible children elements which are input element containers

Peek 2019-04-19 16-32

### PR DESCRIPTION
**Pull request by jeanp413**
Fixes #72626

### CODE REVIEW

File: src/vs/workbench/contrib/files/browser/views/explorerView.ts [hunk 1]
```diff
@@ -176,7 +176,13 @@ export class ExplorerView extends ViewletPanel {
 			const isEditing = !!this.explorerService.getEditableData(e);
 
 			if (isEditing) {
-				await this.tree.expand(e.parent!);
+				// Don't expand when parent is the root in single folder
+				// Fixes https://github.com/Microsoft/vscode/issues/72626
+				const isSingleFolder = this.contextService.getWorkbenchState() === WorkbenchState.FOLDER;
```
Comment (isidorn) [inline src/vs/workbench/contrib/files/browser/views/explorerView.ts]: Use this.tree.getInput() to check if the e.parent is shown or not.
Also use e.parent.isRoot()
Try not to use the exclamation mark to fool the ts compiler.

**Review by isidorn — CHANGES_REQUESTED** [review]

**Comment by isidorn** [comment]
This trully fixes that issue.
However I have left some comments in your changes. Can you please address them.
Also why does this fix this issue? How did you find that this is the root cause?

**Comment by jeanp413** [comment]
@isidorn Looking at this in more detail I found that the real problem is when [disposing the input element](https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/contrib/files/browser/views/explorerViewer.ts#L254-L259) because the [renderElement method](https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/contrib/files/browser/views/explorerViewer.ts#L146) gets called multiple times between `this.tree.expand(e.parent!)`, `this.refresh(false, e.parent);` and `this.tree.reveal(e);`,  and the `ignoreDisposeAndBlur` flag is still turned on because of the 100ms timeout. I'll update my PR with a better fix.

**Comment by jeanp413** [comment]
This seems to be working except that it breaks when you are editing an element and click on the Refresh or Collapse All buttons, maybe they should be disabled like the New File and New Folder buttons.
Also I think `ignoreDisposeAndBlur` flag is no longer needed.

**Comment by isidorn** [comment]
This is good work. Let's merge it in. I will remove the ignoreDisposeAndBlur and check the refresh action. Thanks a lot!

**Comment by jeanp413** [comment]
Great! Don't forget about the `Collapse All` action, it should also be disabled while editing.

**Comment by isidorn** [comment]
`Collapse All` does not break anything for me. So I left it enabled.

**Comment by jeanp413** [comment]
@isidorn I just build and tested it from source code, it breaks.
collapseall

**Comment by isidorn** [comment]
You are correct. I have tackled that via https://github.com/microsoft/vscode/commit/0991cb526e878ec1ed4b17c4e99a451042092fab
Thanks!

**Pull request merged**